### PR TITLE
Remove legacy IDiagnosticService

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
The only client even listening for the events from this is TS.  Note: they are moving to LSP pull diags.  So i'm goign to wait on them to get that in, then remove all of this.  Tagging @MariaSolOs as well.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs [hunk 1]
```diff
@@ -32,11 +32,10 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.UserDiagnos
     [UseExportProvider]
     public class DiagnosticAnalyzerDriverTests
     {
-        private static readonly TestComposition s_compositionWithMockDiagnosticUpdateSourceRegistrationService = EditorTestCompositions.EditorFeatures
-            .AddExcludedPartTypes(typeof(IDiagnosticUpdateSourceRegistrationService))
-            .AddParts(typeof(MockDiagnosticUpdateSourceRegistrationService));
+        private static readonly TestComposition s_compositionWithMockDiagnosticUpdateSourceRegistrationService = EditorTestCompositions.EditorFeatures;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs:35]: note: the names of this compositions need to change.

File: src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs [hunk 2]
```diff
@@ -32,11 +32,10 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.UserDiagnos
...
 
         [Fact]
+
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs:38]:
```suggestion
```

File: src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptDiagnosticService.cs [hunk 3]
```diff
@@ -3,25 +3,21 @@
...
+using Microsoft.CodeAnalysis.Host.Mef;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript;
 
 [Export(typeof(IVSTypeScriptDiagnosticService)), Shared]
 [method: ImportingConstructor]
 [method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class VSTypeScriptDiagnosticService(IDiagnosticService service) : IVSTypeScriptDiagnosticService
+internal sealed class VSTypeScriptDiagnosticService() : IVSTypeScriptDiagnosticService
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptDiagnosticService.cs:19]: this is the only place that actually takes the events from IDiagnosticService and forwards them along.  But it's only for "push style" diagnostics, which we've moved off of, and which TS also definitely does not want.  @MariaSolOs for awareness.  this should go nicely with yoru move to LSP pull diags.

File: src/EditorFeatures/Test/CodeFixes/CodeFixServiceTests.cs [hunk 4]
```diff
@@ -49,7 +47,6 @@ public async Task TestGetFirstDiagnosticWithFixAsync()
 ";
             using var workspace = TestWorkspace.CreateCSharp(code, composition: s_compositionWithMockDiagnosticUpdateSourceRegistrationService, openDocuments: true);
 
-            Assert.IsType<MockDiagnosticUpdateSourceRegistrationService>(workspace.GetService<IDiagnosticUpdateSourceRegistrationService>());
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/CodeFixes/CodeFixServiceTests.cs:52]: tons of asserts taht tests were actually not even using actual product code...

File: src/EditorFeatures/Test/EditAndContinue/RemoteEditAndContinueServiceTests.cs [hunk 5]
```diff
@@ -39,8 +39,7 @@ private static string Inspect(DiagnosticData d)
                 (!string.IsNullOrWhiteSpace(d.DataLocation.UnmappedFileSpan.Path) ? $" {d.DataLocation.UnmappedFileSpan.Path}({d.DataLocation.UnmappedFileSpan.StartLinePosition.Line}, {d.DataLocation.UnmappedFileSpan.StartLinePosition.Character}, {d.DataLocation.UnmappedFileSpan.EndLinePosition.Line}, {d.DataLocation.UnmappedFileSpan.EndLinePosition.Character}):" : "") +
                 $" {d.Message}";
 
-        [Theory]
-        [CombinatorialData]
+        [Theory, CombinatorialData, Obsolete]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/EditAndContinue/RemoteEditAndContinueServiceTests.cs:42]: Obsolete needed as this tests the obsolete constructor of EnCUpdateSource

**Comment by CyrusNajmabadi** [comment]
@mavasani ptal.

**Comment by CyrusNajmabadi** [comment]
@mavasani ptal

**Review by mavasani — APPROVED** [review]

**Pull request merged**